tests/examples/shims/screen.lua

```diff
@@ -85,8 +87,36 @@ function screen._get_extents()
     return xmax.geometry.x+xmax.geometry.width, ymax.geometry.y+ymax.geometry.height
 end
 
+-- The way the `screen` module is used to store both number and object is
+-- problematic since it can cause invalid screens to be "resurrected".
+local function catch_invalid(_, s)
+    -- The CAPI implementation allows `nil`
+    if s == nil or type(s) == "string" then return end
+
+    assert(screen ~= s, "Some code tried (and failed) to shadow the global `screen`")
+
+    -- Valid numbers wont get this far.
+    assert(type(s) == "table", "Expected a table, but got a "..type(s))
+
+    if type(s) == "number" then return end
+
+    assert(s.geometry, "The object is not a screen")
+    assert(s.outputs, "The object is not a screen")
+
+    assert((not screen._deleted[s]) or (not s.valid), "The shims are broken")
+
+    assert(not screen._deleted[s], "The screen "..tostring(s).."has been deleted")
+
+    -- Other errors. If someone place an object in the `screen` module, it wont
+    -- get this far. So the remaining cases are probably bugs.
+    assert(false)
```

Comment:
The last four lines of `assert` are basically dead code, aren't they? This code will end up in `assert(false)` anyway, so is the point just do produce a better error message?

Reply:
> so is the point just do produce a better error message?

Yes